### ISSUE TITLE
Robots not using tools completely

### ISSUE THREAD

**Issue by DoctorToast3** [BuildCraft/BuildCraft#3160]
I think I've discovered a bug where robots won't use Vanilla Minecraft tools completely. The lumberjack robot will take the ax and chop logs until the tool has one block break left on it, then return to its docking station and not continue working until the nearly used tool is taken away from it.

I'm using Buildcraft 7.1.14, Buildcraft-compat 7.1.3, CodeChickCore 1.0.4.29, Iron Chests Mod 1.7.10, Iron Chest Minecarts 1.0.3, Not Enough Items 1.0.3.74, OpenEye 0.6, and Railcraft 9.8.0.0 on Minecraft 1.7.10 and Forge 10.13.4.1558.

**Comment by Kubuxu** [comment]
It was created to allow robots not use-up enchanted tools. Is your axe enchanted?

**Comment by DoctorToast3** [comment]
No it is not, which is why I found it odd. It's just a plain, wooden axe.

**Comment by DoctorToast3** [comment]
Found a way around it! I setup a void pipe with a docking station and gate that requests the nearly broken tools, so the lumberjack gets rid of them!

**Comment by DoctorToast3** [comment]
Or I guess it was just set to accept them, for the record.

**Comment by chemdork123** [comment]
I also saw this with the Knight robots not unloading their nearly-broken swords.  I'll have to try that same workaround, @DoctorToast3 .  You just took a damaged normal wooden axe and set the docking station to accept the nearly wooden axes?

**Comment by DoctorToast3** [comment]
Correct. I set down a single section of void pipe and put a docking station and quartz gate on it. Filtered the gate to accept nearly broken wooden axes, and the little guy started going on his own again!

Sent from my Windows Phone
________________________________

**Comment by chemdork123** [comment]
It doesn't seem to work with Knight robots at all - they are not discarding their sword no matter what options I give them.  Is this an issue for you too?  Lumberjacks have no problem dropping items.

**Comment by Leytek** [comment]
It's not a bug, it's a feature. It allows me automated recycle "broken" iron axes into iron ingots and craft new axes.

**Comment by AEnterprise** [comment]
maybe some kind of setting on the robot to let the player chose if he wants the robot to use up the axe or not? could be as simple as installing a redstone chipset on it to toggle the option

**Comment by AEnterprise** [comment]
what are we gona do about this, in some cases people would want the robot to not completely use up the tool (if it's enchanted and they want to repair it) but in other cases (most of them i guess acutally) they would want the robot to use them up.

dupe bug mentioned above is not robot specific, the mod that allows recycling them should check the damage and not give ingots if it's taken damage

**Comment by AlexIIL** [comment]
Yeah its annoying to deal with. I'm considering changing robots so they can only equip special "robot tools" so that we can make this well defined. And then we don't have to worry about making robots work perfectly with every modded tool.

**Comment by AEnterprise** [comment]
that would be a great compromise, an idea to consider then while on this topic: over the years (again a few days ago even) there has been multiple discussions about having the quarry use a pickaxe or something (usually involving enchants and OP things)

but it might not be a bad idea to have the quarry also use a similar (or the same) tool

finaly: you can change the title of an issue so you might wana change it to something about those robot tools to make it clear

**Comment by dj3520** [comment]
If the "robot tools" repair with energy then this entire issue is gone, in which case I support.

**Comment by AlexIIL** [comment]
Robot tools would be slightly more expensive versions of vanilla tools, with (possibly) less durability. They would not take enchantments either, so they would be used up fully with no issues.

**Comment by chemdork123** [comment]
I suppose you could also adjust it so that any tool with an enchantment is not used up (so you can repair it), and any other tool is used up completely.  I say that because I'm pretty sure outside the enchanted tools, it's not realistic to keep repairing a vanilla un-enchanted tool.
